dopamine/discrete_domains/run_experiment.py

```diff
@@ -625,9 +626,18 @@ def _run_one_iteration(self, iteration):
       iteration: int, current iteration number, used as a global_step for saving
         Tensorboard summaries
     """
-    statistics = super(AsyncRunner, self)._run_one_iteration(iteration)
+    statistics = iteration_statistics.IterationStatistics()
+    tf.logging.info('Starting iteration %d', iteration)
+    num_episodes_train, average_reward_train = self._run_train_phase(
+        statistics)
+    num_episodes_eval, average_reward_eval = self._run_eval_phase(
+        statistics)
     with self._output_lock:
-      self._log_experiment(iteration, statistics)
-      self._checkpoint_experiment(iteration)
+      self._log_experiment(self._completed_iteration, statistics)
+      self._checkpoint_experiment(self._completed_iteration)
+      self._save_tensorboard_summaries(
+          self._completed_iteration, num_episodes_train, average_reward_train,
+          num_episodes_eval, average_reward_eval)
+      self._completed_iteration += 1
```

Comment:
Should this be at the beginning of the `with` block?

Reply:
`iteration` was starting at `0`. Here I wanted to maintain that so the increment has to go at the end? Is that fine?

Reply:
Sorry.  I think it's fine as it is.  Asynchronous execution is hard to think about, and when I made this comment I missed that both incrementing and writing is done in a different block than the execution.